### DISCUSSION TITLE
When switching from TriLib to GLTFast, all prefab (glb) meshes go missing. What to do?

### DISCUSSION THREAD

**Discussion by JPhilipp** [atteneder/glTFast#492] [Q&A]
Hi, I'm posting here instead of in Issues as this seems to be more of a Unity issue, rather than a GLTfast issue, but I could be wrong/ please feel free to move if wanted.

So the issue is that I have created a bunch of prefab scenes, and they use gameObjects with mesh filters which have *.glb meshes. I did this while still using the TriLib asset, which also loads GLTF. Now I wanted to switch to GLTFast, but after I did (and removed TriLib), the following happened:
- Unity changed all meta files of glb files, as shown in the example below.

![Trilib-Issue](https://user-images.githubusercontent.com/1754503/194148963-88d14b86-cfd8-4283-8977-516c84a42641.png)

- All my prefabs now showed the meshes as missing, i.e. they became unusable. The prefab themselves didn't change, it seems it's just something like a guid mixup. It is not, however, the top guid in the glb meta file that is changed, as the screenshot above shows.
- The glb files when looked at or individually dragged into the scene still load fine using GLTFast, there's no problem there.

A workaround remedy for above is this:
1. Put back TriLib into the project.
2. Use script settings define GLTFAST_FORCE_DEFAULT_IMPORTER_OFF (even though we want GLTFast, but this is to avoid the message about conflicting importers).
3. Discard all Meta file changes via Git.
4. Open Unity again -- the meshes should now be back.

Is there something better to do? And why are the meshes missing in the first place? I came across [this article](https://www.boristhebrave.com/2020/02/05/messing-with-unitys-guids/), but not sure if related or a red herring. Thanks!

**Comment by atteneder** [comment]
Hi,

Thanks for bringing this up. I never thought about conversion workflows before.

Did you drag the main asset into the scene? Or did you drag sub-content (like other glTF scenes or single gameobjects)? The latter probably won't work, as the asset's content is different for both importers.

I just successfully tested switching between UnityGLTF and glTFast (that is switching importers on a prefab without any overrides).

I'm not surprise the GUID on line 10 changed, as I assume it's the GUID of the used script (trilib or gltfast). The GUID of the actual prefab is on line 2 and stayed constant!

Please try this:

- Install both Trilib and glTFast
- Enable the scripting define `GLTFAST_FORCE_DEFAULT_IMPORTER_OFF`
- Try switching the importer from TriLib to glTFast:

<img width="387" alt="image" src="https://user-images.githubusercontent.com/2593014/194279295-5358c745-3e62-41ae-8af4-10014d5df0f9.png">

This is what worked for me.

**Comment by JPhilipp** [comment]
Thanks for the reply!

The moment I do as you say -- install both Trilib and GLTFast, use GLTFAST_FORCE_DEFAULT_IMPORTER_OFF, try switching -- the mesh in my prefab again disappears (it'll show as a missing reference). See before and after below -- the mesh in question is the table:

![image](https://user-images.githubusercontent.com/1754503/194298195-1ec724b8-3063-4285-a6a7-caa4bb3c8aa0.png)

![image](https://user-images.githubusercontent.com/1754503/194298314-55ea432d-6b9c-48be-bba3-ad7aedab520f.png)

(For reference, I'm working on a GPT-3-based world creator, the green cubes are reserved item spots for things to be put on or under the table.)

As per a Git check, no single file except the *.glb.meta of the asset was changed (and as before, the top guid in it remained the same, it was only the script guid and other script info in the meta file that changed).

The mesh itself still works using GLTFast as it generates a preview and can be dragged into the scene -- and if I wanted to, I could manually reassociate it by dragging it onto the prefab's mesh spot. The issue is that I have plenty of furniture prefabs already set up, and wanted to see if there's a non-workaround way to fix it.

On the other hand, your approach of switching mesh by mesh manually actually helps me do a manual workaround, so thanks!

**Reply by atteneder** [reply]
Hmm, sorry to say I'm clueless, but glad the workaround has other benefits for you.
